cmake/CMakeLists.txt

```
@@ -580,30 +603,21 @@ if(BUILD_LIB)
 
   set_target_properties(lammps PROPERTIES OUTPUT_NAME lammps${LAMMPS_LIB_SUFFIX})
   set_target_properties(lammps PROPERTIES SOVERSION ${SOVERSION})
-  install(TARGETS lammps LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  install(TARGETS lammps EXPORT LAMMPS_Targets LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  target_include_directories(lammps PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
```

Could you add the lammps folder directly to the include path?
```suggestion
  target_include_directories(lammps PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>/lammps)
```
This way `add_subdirectory` and `find_package` both expose the headers with the same relative path.

---

Well, but the header are supposed to be included as `lammps/lammps.h` and not just `lammps.h`. Some lammps have really generic names, e.g. `pointer.h`, so don't want that without the `lammps/` prefix.

---

I don't have an opinion on this, i'd just prefer if both methods of inclusion could be used interchangeably.

---

@akohlmey any opinion?

---

i still don't know what @sgsaenger is trying to achieve and thus am not convinced that whatever he is asking for is something that has to be addressed on the LAMMPS side.
Also the priority is to make compiling, installing, developing,  and using LAMMPS **directly** easy and consistent.

That said, and as I already mentioned, we will have a discussion with the core LAMMPS developers about what headers we **want** to export for using the C++ interface. Historically, the expectation was that people writing tools on top of LAMMPS would not do this based on a pre-packaged, pre-compiled, pre-installed version of LAMMPS, but adapt their own code to access the LAMMPS sources in place. Using CMake and the rapidly growing adoption of pre-compiled and pre-packages LAMMPS is something new and thus there is no fixed opinion of what to do and how to do it.

---

> i still don't know what @sgsaenger is trying to achieve and thus am not convinced that whatever he is asking for is something that has to be addressed on the LAMMPS side.

> Historically, the expectation was that people writing tools on top of LAMMPS would not do this based on a pre-packaged, pre-compiled, pre-installed version of LAMMPS, but adapt their own code to access the LAMMPS sources in place.

This was my basic goal, use lammps as a library by embedding it in my build tree (i.e. using `add_subdirectory` in cmake). The usual _CMake_ way of doing things is using installed packages via `find_package`, which is now also working thanks to @junghans.
With the increasing adoption of modern cmake, git + submodules and header-only libraries, there's now this pattern of
```
find_package(LAMMPS QUIET)
if(NOT LAMMPS_FOUND)
  add_subdirectory(lammps)
endif()
```
where the build defaults to using an installed version of a package but falls back to an embedded one (you're also already using this in e.g. the `COUPLE/simple` example). This doesn't work though if `find_package` enforces `#include <lammps/lammps.h>` but `add_package` requires `#include <lammps.h>`.
Both are fine by me, the former is more clean and i understand @junghans pushing for it, the latter is easier to implement because it doesn't require changes to the lammps source tree.

---

Note @junghans:
the `mpi_stubs` target forwards the include path `$<INSTALL_PATH:${CMAKE_INSTALL_INCLUDEDIR}/lammps>`. Probably necessary as enforcing a using code to `#include <lammps/mpi.h>` is way less useful than having a dropin replacement for the usual include, but it defies your intention and adds another inconsistency.